antiope-sns/src/Antiope/SNS.hs

```diff
@@ -10,13 +10,12 @@ import Data.Text       (Text)
 import Network.AWS     (MonadAWS, send)
 import Network.AWS.SNS
 
-publishMsg :: MonadAWS m => Text -> Text -> m ()
+publishMsg :: MonadAWS m => Text -> Text -> m (Maybe Text)
 publishMsg topicArn message = do
-  _ <- send $ publish message & pTopicARN ?~ topicArn
-  return ()
+  resp <- send $ publish message & pTopicARN ?~ topicArn
+  return $ resp ^. prsMessageId
 
-subscribeMsg :: MonadAWS m => Text -> Text -> Text -> m ()
+subscribeMsg :: MonadAWS m => Text -> Text -> Text -> m (Maybe Text)
```

Comment:
1) Please don't do `Msg`
2) The function  doesn't do anything with messages. It _looks like_ it subscribes topic to the endpoint, so it doesn't need to have `Message` it  its name.
3) `Text`s are very  confusing. Consider  `TopicArn -> Protocol -> Endpoint -> m (Maybe SubscriptionArn)` or whatever these `Endpoint` and `Protocol` actually mean.